Not matching distribution found for tensorflow=1.1.0 (from auDeep 0.9.1)

### System information
- **OS Platform and Distribution**: Windows 7 Professional
- **auDeep version (Git commit SHA / release version)**: auDeep 0.9.1
- **Python version**: Python 3.6.5, Anaconda3 5.2.0
- **Exact command to reproduce**: pip install ./auDeep

### Describe the problem
I'm following the exact steps in the README to install auDeep, but when writting: 
`pip install ./auDeep`

I get the following error:

> Collecting tensorflow==1.1.0 (from auDeep==0.9.1)
>   Could not find a version that satisfies the requirement tensorflow==1.1.0 (fro
> m auDeep==0.9.1) (from versions: 1.2.0rc2, 1.2.0, 1.2.1, 1.3.0rc0, 1.3.0rc1, 1.3
> .0rc2, 1.3.0, 1.4.0rc0, 1.4.0rc1, 1.4.0, 1.5.0rc0, 1.5.0rc1, 1.5.0, 1.5.1, 1.6.0
> rc0, 1.6.0rc1, 1.6.0, 1.7.0rc0, 1.7.0rc1, 1.7.0, 1.7.1, 1.8.0rc0, 1.8.0rc1, 1.8.
> 0, 1.9.0rc0, 1.9.0rc1, 1.9.0rc2, 1.9.0, 1.10.0rc0, 1.10.0rc1, 1.10.0, 1.11.0rc0,
>  1.11.0rc1, 1.11.0rc2, 1.11.0, 1.12.0rc0, 1.12.0rc1, 1.12.0rc2, 1.12.0, 1.13.0rc
> 0, 1.13.0rc1, 1.13.0rc2, 1.13.1, 2.0.0a0)
> No matching distribution found for tensorflow==1.1.0 (from auDeep==0.9.1)

I have read that tensorflow 1.1.0 is no longer supported by pip, could that be the problem?

As a temporal solution, I tried downloading tensorflow 1.1.0 .whl for windows, but the only file I could find was for Python 3.5, so I installed Python 3.5 and tried manually installing the tensorflow .whl. And then another problem arised, when writting the same auDeep installation command, the output was:

> yield inside async function python

And after checking this, I found out that 'yield' for asynchronous generators could only be used after Python 3.6.

Does anyone have a solution for this or has faced a similar problem?

Thanks in advance

Thank you! It does solve that issue, although I get another error now.

After:

`pip install ./auDeep`

I get the following message: 

> Successfully installed PrettyTable-0.7.2 PyYAML-5.1 absl-py-0.7.1 astor-0.7.1 at trs-19.1.0 auDeep-0.9.2 cffi-1.12.2 cftime-1.0.3.4 cliff-2.14.1 cmd2-0.9.11 colo rama-0.4.1 cycler-0.10.0 gast-0.2.2 grpcio-1.19.0 h5py-2.9.0 keras-applications- 1.0.7 keras-preprocessing-1.0.9 kiwisolver-1.0.1 liac-arff-2.4.0 markdown-3.1 ma tplotlib-3.0.3 mock-2.0.0 netCDF4-1.5.0.1 numpy-1.16.2 pandas-0.24.2 pbr-5.1.3 p rotobuf-3.7.1 pycparser-2.19 pyparsing-2.4.0 pyperclip-1.7.0 pyreadline-2.1 pyso undfile-0.9.0.post1 python-dateutil-2.8.0 pytz-2019.1 scikit-learn-0.20.3 scipy- 1.2.1 six-1.12.0 sklearn-0.0 stevedore-1.30.1 tensorboard-1.13.1 tensorflow-1.13 .1 tensorflow-estimator-1.13.0 termcolor-1.1.0 wcwidth-0.1.7 werkzeug-0.15.2 xar ray-0.10.0 


But there is another problem. I thought no patch for Tensorflow would be necessary as the version is 1.13.1 instead of 1.1.0, but when trying to check auDeep version:
`auDeep --version`

The following stack trace appears:

> [ERROR] audeep - Traceback (most recent call last):
>   File "c:\audeep_virtualenv\lib\site-packages\tensorflow\python\pywrap_tensorfl
> ow.py", line 58, in <module>
>     from tensorflow.python.pywrap_tensorflow_internal import *
>   File "c:\audeep_virtualenv\lib\site-packages\tensorflow\python\pywrap_tensorfl
> ow_internal.py", line 28, in <module>
>     _pywrap_tensorflow_internal = swig_import_helper()
>   File "c:\audeep_virtualenv\lib\site-packages\tensorflow\python\pywrap_tensorfl
> ow_internal.py", line 24, in swig_import_helper
>     _mod = imp.load_module('_pywrap_tensorflow_internal', fp, pathname, descript
> ion)
>   File "c:\audeep_virtualenv\lib\imp.py", line 243, in load_module
>     return load_dynamic(name, filename, file)
>   File "c:\audeep_virtualenv\lib\imp.py", line 343, in load_dynamic
>     return _load(spec)
> ImportError: DLL load failed with error code -1073741795
> Failed to load the native TensorFlow runtime.
> See https://www.tensorflow.org/install/errors
> for some common reasons and solutions.  Include the entire stack trace
> above this error message when asking for help.

I thought it could be due to the necessary patch for Tensorflow within the virtualenv, so I installed `cygwin `and its `patch `package and when writting on cygwin:


`$ patch ./audeep_virtualenv/Lib/site-ackages/tensorflow/python/framework/meta_graph.py ./auDeep/patches/fix_import_bug.patch`

I get the following message:

> patching file ./audeep_virtualenv/Lib/site-packages/tensorflow/python/framework/meta_graph.py
> Hunk #1 FAILED at 472 (different line endings).
> 1 out of 1 hunk FAILED -- saving rejects to file ./audeep_virtualenv/Lib/site-packages/tensorflow/python/framework/meta_graph.py.rej

So I guess auDeep is not completely installed, any ideas about how to solve the issue?

Thanks again in advance!

--- a/README.md
+++ b/README.md
@@ -27,8 +27,8 @@ The minimal requirements to install auDeep are listed below.
 ### Additional Requirements for GPU Support
 The `setup.py` script automatically checks if a compatible CUDA version is available on the system. If GPU support is desired, make sure that the following dependencies are installed **before** installing auDeep.
 
-- CUDA Toolkit 8.0
-- cuDNN 5.1 (optional)
+- CUDA Toolkit 10.0
+- cuDNN 7.4.1
 
 By default, the CUDA libraries are required to be available on the system path (which should be the case after a standard installation). If, for some reason, the `setup.py` script fails to detect the correct CUDA version, consider manually installing `tensorflow-gpu` 1.1.0 prior to installing auDeep.
 
@@ -42,7 +42,7 @@ These Python packages are installed automatically during setup by `pip`, and are
 - pandas
 - scipy
 - sklearn
-- tensorflow or tensorflow-gpu 1.1.0
+- tensorflow or tensorflow-gpu 1.13.0
 - xarray
 
 ## Installing in a virtualenv
@@ -79,14 +79,6 @@ If everything went well, you should see `(audeep_virtualenv)` prepended to the c
 ```
 where `auDeep` is the name of the project root directory containing the `setup.py` file. This will fetch all required depencendies and install them in the virtualenv.
 
-Once installation is complete, the TensorFlow installation within the virtualenv needs to be patched, in order to fix a bug that has not yet been resolved in the official repositories. The required changes are provided as a GNU patch file at `patches/fix_import_bug.patch` below the project root directory. To apply the patch the GNU patch utility is required, which should be installed by default on Linux systems. On Windows, it can be obtained, for example, through `cygwin`. Please note that you have to manually select the `patch` package during installation of `cygwin`, as it is not installed by default.
-```
-Linux:
-> patch audeep_virtualenv/lib/python3.5/site-packages/tensorflow/python/framework/meta_graph.py auDeep/patches/fix_import_bug.patch
-
-Windows:
-> patch .\audeep_virtualenv\Lib\site-packages\tensorflow\python\framework\meta_graph.py .\auDeep\patches\fix_import_bug.patch
-```
 This completes installation, and the toolkit CLI can be accessed through the `audeep` binary.
 ```
 > audeep --version

--- a/audeep/backend/training/base.py
+++ b/audeep/backend/training/base.py
@@ -321,11 +321,10 @@ class BaseFeatureLearningWrapper(LoggingMixin):
             The Tensorflow graph deserialized from the specified file, with any unbound inputs bound to the respective
             tensors defined in the `input_map` parameter
         """
-        input_map = {"$unbound_inputs_%s" % k: input_map[k] for k in input_map}
+        input_map = {k: input_map[k] for k in input_map}
 
         saver = tf.train.import_meta_graph(str(model_filename.with_suffix(".meta")),
-                                           clear_devices=True,
-                                           import_scope="model",
+                                           clear_devices=True, import_scope=None,
                                            input_map=input_map)
 
         return GraphWrapper(graph=tf.get_default_graph(),
@@ -352,16 +351,18 @@ class BaseFeatureLearningWrapper(LoggingMixin):
         kwargs: keyword arguments
             Additional keyword arguments specifying the model architecture
         """
-        with tf.Graph().as_default():
+        with tf.Graph().as_default() as g:
+
+
             tf_inputs = tf.placeholder(name="inputs",
                                        shape=[feature_shape[0], None, feature_shape[1]],
                                        dtype=tf.float32)
 
             self._create_model(tf_inputs, **kwargs)
 
+
             tf.train.export_meta_graph(filename=str(model_filename.with_suffix(".meta")),
-                                       export_scope="model",
-                                       clear_devices=True)
+                                       clear_devices=True, export_scope=None)
 
     def train_model(self,
                     model_filename: Path,

--- a/setup.py
+++ b/setup.py
@@ -5,7 +5,7 @@ from subprocess import CalledProcessError, check_output
 from setuptools import setup, find_packages
 
 PROJECT = "auDeep"
-VERSION = "0.9.1"
+VERSION = "0.9.2"
 LICENSE = "GPLv3+"
 AUTHOR = "Michael Freitag"
 AUTHOR_EMAIL = "freitagm@fim.uni-passau.de"
@@ -42,7 +42,7 @@ if not tensorflow_found:
             major = int(version_string.group("major"))
             minor = int(version_string.group("minor"))
 
-            if major != 8:
+            if major != 10 or minor != 0:
                 print("detected incompatible CUDA version %d.%d" % (major, minor))
             else:
                 print("detected compatible CUDA version %d.%d" % (major, minor))
@@ -56,9 +56,9 @@ if not tensorflow_found:
         print("error during CUDA detection: %s", e)
 
     if use_gpu:
-        dependencies.append("tensorflow-gpu==1.1.0")
+        dependencies.append("tensorflow-gpu==1.13.1")
     else:
-        dependencies.append("tensorflow==1.1.0")
+        dependencies.append("tensorflow==1.13.1")
 else:
     print("tensorflow already installed, skipping CUDA detection")
 

